syntax = "proto3";

message Person {
    int32 id = 1;
    string first_name = 2;
    string last_name = 3;
    string company_name = 4;
}


message PersonList {
    repeated Person items = 1;
}


message Empty {
}

message CreatePersonRequest {
    string first_name = 1;
    string last_name = 2;
    string company_name = 3;
}

message GetPersonRequest {
    int32 id = 1;
}

service PersonService {
    rpc GetAll(Empty) returns (PersonList);
    rpc Get(GetPersonRequest) returns (Person);
    rpc Create(CreatePersonRequest) returns (Person);
}